pallets/stream-payment/src/lib.rs

```diff
@@ -0,0 +1,907 @@
+            T::Assets::decrease_deposit(&stream.config.asset_id, &stream.source, stream.deposit)?;
+
+            // Remove stream from storage.
+            Streams::<T>::remove(stream_id);
+            LookupStreamsWithSource::<T>::remove(stream.source, stream_id);
+            LookupStreamsWithTarget::<T>::remove(stream.target, stream_id);
+
+            // Emit event.
+            Pallet::<T>::deposit_event(Event::<T>::StreamClosed {
+                stream_id,
+                refunded: stream.deposit,
+            });
+
+            Ok(().into())
+        }
+
+        /// Perform the pending payment of a stream. Anyone can call this.
+        #[pallet::call_index(2)]
+        pub fn perform_payment(
+            origin: OriginFor<T>,
+            stream_id: T::StreamId,
+        ) -> DispatchResultWithPostInfo {
+            // No problem with anyone updating any stream.
+            let _ = ensure_signed(origin)?;
+
+            let mut stream = Streams::<T>::get(stream_id).ok_or(Error::<T>::UnknownStreamId)?;
+            Self::perform_stream_payment(stream_id, &mut stream)?;
+            Streams::<T>::insert(stream_id, stream);
+
+            Ok(().into())
+        }
+
+        /// Requests a change to a stream config or deposit.
+        ///
+        /// If the new config don't change the time unit and asset id, the change will be applied
+        /// immediately if it is at the desadvantage of the caller. Otherwise, the request is stored
+        /// in the stream and will have to be approved by the other party.
+        ///
+        /// This call accepts a deposit change, which can only be provided by the source of the
+        /// stream. An absolute change is required when changing asset id, as the current deposit
+        /// will be released and a new deposit is required in the new asset.
+        #[pallet::call_index(3)]
+        pub fn request_change(
+            origin: OriginFor<T>,
+            stream_id: T::StreamId,
+            kind: ChangeKind<T::Balance>,
+            new_config: StreamConfigOf<T>,
+            deposit_change: Option<DepositChange<T::Balance>>,
+        ) -> DispatchResultWithPostInfo {
+            let origin = ensure_signed(origin)?;
+            let mut stream = Streams::<T>::get(stream_id).ok_or(Error::<T>::UnknownStreamId)?;
+
+            let requester = stream
+                .account_to_party(origin)
+                .ok_or(Error::<T>::UnauthorizedOrigin)?;
+
+            ensure!(
+                requester == Party::Source || deposit_change.is_none(),
+                Error::<T>::TargetCantChangeDeposit
+            );
+
+            if stream.config == new_config && deposit_change.is_none() {
+                return Ok(().into());
+            }
+
+            // If asset id and time unit are the same, we allow to make the change
+            // immediatly if the origin is at a disadvantage.
+            // We allow this event if there is already a pending request.
+            'immediate: {
```

Comment:
You can avoid this weird control flow by refactoring the condition into a function:

```suggestion
            if allow_immediate_config_change(&stream, &new_config, &requester) {
```

Reply:
Yes please